### PR TITLE
Fix system path checks

### PR DESCRIPTION
**Pull request by tobeyStraitjacket**
Fixes an issue experienced by a user who had Subnautica installed in the following path: `E:\Windows Steam Library\steamapps\common\Subnautica\`

Caused by slightly overzealous file IO caution

### CODE REVIEW

File: Data/latest-version.txt [hunk 1]
```diff
@@ -1 +1 @@
-4.0.2.3
+4.0.3.0
```
Comment (AlexejheroYTB) [inline Data/latest-version.txt:1]: 4.0.2.4?

File: QModManager/Patching/Patcher.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ internal static string QModBaseDir
         {
             get
             {
-                if (Environment.CurrentDirectory.Contains("system32") && Environment.CurrentDirectory.Contains("Windows"))
+                if (Environment.CurrentDirectory.ToLowerInvariant().Contains("system32") && Environment.CurrentDirectory.ToLowerInvariant().Contains("windows"))
```
Comment (AlexejheroYTB) [inline QModManager/Patching/Patcher.cs:19]: This hasn't happened in a while now and the check can be removed without issues.

File: QModPluginEmulator/QModPluginGenerator.cs [hunk 3]
```diff
@@ -150,7 +150,7 @@ private static void SaveQMMAssemblyCache()
 
         private static void ClearBepInExCache()
         {
-            if (BepInExCachePath.Contains("system32") || BepInExCachePath.Contains("Windows"))
+            if (BepInExCachePath.ToLowerInvariant().Contains("system32") && BepInExCachePath.ToLowerInvariant().Contains("windows"))
```
Comment (AlexejheroYTB) [inline QModPluginEmulator/QModPluginGenerator.cs:153]: Same here ^

File: Data/latest-version.txt (hunk 1 above)
Comment (tobeyStraitjacket) [reply Data/latest-version.txt:1]: That's fine, there's another patch that isn't released yet that is 4.0.2.3 which is fixing some install issues, so we can probably actually just keep 4.0.2.3

**Review by PrimeSonic — APPROVED** [review]
I see no reason not to include this.

@AlexejheroYTB any comments?

**Comment by AlexejheroYTB** [comment]
The checks were basically put in place in QMM1 when the QMods path was sometimes System32 for whatever reason. That no longer happens for another whatever reason, so the checks dont need to be there.

**Review by AlexejheroYTB — APPROVED** [review]

**Pull request merged**